activerecord/lib/active_record/associations/belongs_to_association.rb

```diff
@@ -107,8 +107,7 @@ def target_id
         end
 
         def stale_state
-          result = owner._read_attribute(reflection.foreign_key)
-          result && result.to_s
```

Comment:
I would keep this as a separate line. This change would cause `nil` to be turned into a string, which I believe could potentially cause extra queries.